packages/legend-graph/src/helpers/DomainHelper.ts

```diff
@@ -49,3 +59,270 @@ export const getMilestoneTemporalStereotype = (
   });
   return stereotype;
 };
+
+export const milestoningPropertyGenerator = (
+  propertyOwner: PropertyOwner,
+  graph: PureModel,
+): void => {
+  propertyOwner.properties.forEach((property) => {
+    if (property.genericType.value.rawType instanceof Class) {
+      const milestonedStereotype = getMilestoneTemporalStereotype(
+        property.genericType.value.rawType,
+        graph,
+      );
+      switch (milestonedStereotype) {
+        case MILESTONING_STEROTYPES.BUSINESS_TEMPORAL: {
+          const dateProperty = new DerivedProperty(
+            `${property.name}`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONE),
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                graph.getClass(property.genericType.value.rawType.path),
+              ),
+            ),
+            property.owner,
+          );
+          dateProperty.parameters = [
+            {
+              _type: 'var',
+              class: 'Date',
+              name: DEFAULT_MILESTONING_PARAMETERS.BUSINESS_DATE,
+              multiplicity: {
+                lowerBound: 1,
+                upperBound: 1,
+              },
+            },
+          ];
+          dateProperty.body = [
+            {
+              _type: 'property',
+              parameters: [
+                {
+                  _type: 'var',
+                  name: 'this',
+                },
+              ],
+              property: `${property.name}`,
+            },
+          ];
+          const milestonedAllVersions = new Property(
+            `${property.name}AllVersions`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONEMANY),
+            GenericTypeExplicitReference.create(
+              new GenericType(
+                graph.getClass(property.genericType.value.rawType.path),
+              ),
+            ),
+            property.owner,
+          );
+          const milestonedAllVersionsInRange = new DerivedProperty(
+            `${property.name}AllVersionsInRange`,
+            graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ONE),
```

Comment:
should this also be `ONEMANY` MULTIPLICITY ?

Reply:
Just realized that multiplicity of all these generated properties is dependent on the actual milestoning property. Will fix that